internal/hcs/system.go

```diff
@@ -281,39 +276,37 @@ func (computeSystem *System) ID() string {
 
 // Shutdown requests a compute system shutdown, if IsPending() on the error returned is true,
 // it may not actually be shut down until Wait() succeeds.
-func (computeSystem *System) Shutdown() (err error) {
+func (computeSystem *System) Shutdown() (_ bool, err error) {
 	computeSystem.handleLock.RLock()
 	defer computeSystem.handleLock.RUnlock()
 
 	operation := "hcsshim::ComputeSystem::Shutdown"
 	computeSystem.logOperationBegin(operation)
 	defer func() {
-		if IsAlreadyClosed(err) || IsAlreadyStopped(err) || IsPending(err) {
-			computeSystem.logOperationEnd(operation, nil)
-		} else {
-			computeSystem.logOperationEnd(operation, err)
-		}
+		computeSystem.logOperationEnd(operation, err)
 	}()
 
 	if computeSystem.handle == 0 {
-		return makeSystemError(computeSystem, "Shutdown", "", ErrAlreadyClosed, nil)
+		return false, nil
 	}
 
 	var resultp *uint16
 	syscallWatcher(computeSystem.logctx, func() {
 		err = hcsShutdownComputeSystem(computeSystem.handle, "", &resultp)
 	})
 	events := processHcsResult(resultp)
-	if err != nil {
-		return makeSystemError(computeSystem, "Shutdown", "", err, events)
+	if IsAlreadyStopped(err) || IsNotExist(err) {
+		return false, nil
 	}
-
-	return nil
+	if err != nil && !IsPending(err) {
+		return false, makeSystemError(computeSystem, "Shutdown", "", err, events)
+	}
+	return true, nil
 }
 
 // Terminate requests a compute system terminate, if IsPending() on the error returned is true,
 // it may not actually be shut down until Wait() succeeds.
-func (computeSystem *System) Terminate() (err error) {
+func (computeSystem *System) Terminate() (_ bool, err error) {
```

Comment:
As we spoke offline. Lets skip `delivered` in this case since the pattern is always safe to call `sys.Shutdown(); sys.Wait()` or `sys.Terminate(); sys.Wait()`